### PR TITLE
Possible fix for 303 - Ignoring legacy Word 97-2003 files

### LINKED ISSUE #303: [Bug] Can't decode DOC files
### Context / Scenario

The user upload a document with the DOC format

### What happened?

When the user upload a document with the legacy DOC format, the application throws an exception saying that the file is corrupt. Analyzing the logs is shown that the application is trying to parse it as a DOCX file.

### Importance

I cannot use Kernel Memory

### Platform, Language, Versions

It happens in Windows and Linux in dotnet 8

### Relevant log output

```shell
System.IO.FileFormatException: File contains corrupted data.
   at System.IO.Packaging.ZipPackage..ctor(Stream s, FileMode packageFileMode, FileAccess packageFileAccess)
   at System.IO.Packaging.Package.Open(Stream stream, FileMode packageMode, FileAccess packageAccess)
   at DocumentFormat.OpenXml.Packaging.PackageLoader.OpenCore(Stream stream, Boolean readWriteMode)
   at DocumentFormat.OpenXml.Packaging.WordprocessingDocument.Open(Stream stream, Boolean isEditable, OpenSettings openSettings)
   at DocumentFormat.OpenXml.Packaging.WordprocessingDocument.Open(Stream stream, Boolean isEditable)
   at Microsoft.KernelMemory.DataFormats.Office.MsWordDecoder.DocToText(Stream data) in /sistemas/projetos/teste-llama/kernel-memory/kernel-memory/service/Core/DataFormats/Office/MsWordDecoder.cs:line 28
   at Microsoft.KernelMemory.DataFormats.Office.MsWordDecoder.DocToText(BinaryData data) in /sistemas/projetos/teste-llama/kernel-memory/kernel-memory/service/Core/DataFormats/Office/MsWordDecoder.cs:line 23
   at Microsoft.KernelMemory.Handlers.TextExtractionHandler.ExtractTextAsync(FileDetails uploadedFile, BinaryData fileContent, CancellationToken cancellationToken) in /sistemas/projetos/teste-llama/kernel-memory/kernel-memory/service/Core/Handlers/TextExtractionHandler.cs:line 140
   at Microsoft.KernelMemory.Handlers.TextExtractionHandler.InvokeAsync(DataPipeline pipeline, CancellationToken cancellationToken) in /sistemas/projetos/teste-llama/kernel-memory/kernel-memory/service/Core/Handlers/TextExtractionHandler.cs:line 78
   at Microsoft.KernelMemory.Pipeline.DistributedPipelineOrchestrator.RunPipelineStepAsync(DataPipeline pipeline, IPipelineStepHandler handler, CancellationToken cancellationToken) in /sistemas/projetos/teste-llama/kernel-memory/kernel-memory/service/Core/Pipeline/DistributedPipelineOrchestrator.cs:line 214
   at Microsoft.KernelMemory.Pipeline.DistributedPipelineOrchestrator.<>c__DisplayClass3_0.<<AddHandlerAsync>b__0>d.MoveNext() in /sistemas/projetos/teste-llama/kernel-memory/kernel-memory/service/Core/Pipeline/DistributedPipelineOrchestrator.cs:line 148
```

### PR DESCRIPTION
**Pull request by amomra**
## Motivation and Context (Why the change? What's the scenario?)

Proposal for fix #303 

## High level description (Approach, Design)

Like described in the issue, the legacy Word 97-2003 files will be ignored. Also the MIME types for the XML formats are changed to reflect the actual content type for those files.

### CODE REVIEW

File: service/Abstractions/Pipeline/MimeTypes.cs [hunk 1]
```diff
@@ -11,9 +11,9 @@ public static class MimeTypes
     public const string PlainText = "text/plain";
     public const string MarkDown = "text/plain-markdown";
     public const string Html = "text/html";
-    public const string MsWord = "application/msword";
-    public const string MsPowerPoint = "application/mspowerpoint";
-    public const string MsExcel = "application/msexcel";
+    public const string MsWord = "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
```
Comment (dluc) [inline service/Abstractions/Pipeline/MimeTypes.cs:14]: at some point we're going to fix the support for old type, could you leave this and introduce a new one for modern Word? Something like 
```
MsWord = "application/msword";
MsWordOpenXML = "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
```

I would apply the same suffix also to the other 2 below -- or leave them untouched until future work that will support both old and new Excel/Powerpoint formats, ie limit the PR scope

**Review by dluc — CHANGES_REQUESTED** [review]

File: service/Abstractions/Pipeline/MimeTypes.cs (hunk 1 above)
Comment (amomra) [reply service/Abstractions/Pipeline/MimeTypes.cs:14]: It is OK to use the X suffix as the old code used it? Also, should I apply the suffix in the decoders as well as seen in this code?

```cs
case MimeTypes.MsWordX:
    this._log.LogDebug("Extracting text from MS Word file {0}", uploadedFile.Name);
    content = new MsWordDecoder().ExtractContent(fileContent);
    break;
```

Comment (dluc) [reply service/Abstractions/Pipeline/MimeTypes.cs]: sure these constants are internal only. 

About the decoder, if you mean TextExtractionHandler yes, I would change `case MimeTypes.MsWord:` to `case MimeTypes.MsWordX:`, so ".doc" files will be caught by the default case and log a warning. 

You could also add a case for `case MimeTypes.MsWord:` and log a more explicit warning, something like `Legacy Word 97 format not supported, it is recommended to migrate to the newer OpenXML format`.

File: service/Abstractions/Pipeline/MimeTypes.cs [hunk 2]
```diff
@@ -68,10 +73,12 @@ public class MimeTypesDetection : IMimeTypeDetection
...
             { FileExtensions.WebPageUrl, MimeTypes.WebPageUrl },
-            { FileExtensions.MsWord, MimeTypes.MsWord },
-            { FileExtensions.MsWordX, MimeTypes.MsWord },
-            { FileExtensions.MsPowerPoint, MimeTypes.MsPowerPoint },
-            { FileExtensions.MsExcel, MimeTypes.MsExcel },
+            // { FileExtensions.MsWord, MimeTypes.MsWord }, // TODO: add support for legacy doc files
+            { FileExtensions.MsWordX, MimeTypes.MsWordX },
+            // { FileExtensions.MsPowerPoint, MimeTypes.MsPowerPoint }, // TODO: add support for legacy ppt files
+            { FileExtensions.MsPowerPointX, MimeTypes.MsPowerPointX },
+            // { FileExtensions.MsExcel, MimeTypes.MsExcel }, // TODO: add support for legacy xls files
```
Comment (dluc) [inline service/Abstractions/Pipeline/MimeTypes.cs:80]: no need to comment out the mappings, they are useful when logging warnings/errors

**Review by dluc — APPROVED** [review]

**Pull request merged**